config/cmake/configGenerator.cmake

```diff
@@ -11,95 +11,36 @@
 if ( MSVC )
     set(WERROR_FLAG "/W4 /WX")
 else( MSVC )
-set(WERROR_FLAG "-Werror")
+    set(WERROR_FLAG "-Werror")
 endif ( MSVC )
 
-
 set(TEST_CXX_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/test_compiler_cxx)
 
+# TODO these shouldn't depend on whether the compiler is in c++17 mode,
+# and should be requested from the compiler (see CMAKE_CXX_KNOWN_FEATURES)
```

Comment:
Not all versions of cmake have the same CMAKE_CXX_KNOWN_FEATURES or even know about C++17 and thus we can't rely on that structure for the supported versions of cmake.    Most of these are CXX_17 features hence if we are not enabling the c++17 features they should not be available regardless.  

You will need to verify this works properly.  I thought there was some reason I had it structured the way I did but I don't remember what it was right.  Something with how configure_file interacted with the variables.

Reply:
Looking into it, CMAKE_CXX_KNOWN_FEATURES isn't useful at all; even in the latest release they're all mostly from the c++11 days.

I hadn't read through CheckLatestCXXStandardOption, it makes more sense why that depends on ENABLE_CXX_17 now.